compactor_scheduler/src/scheduler.rs

```diff
@@ -54,9 +80,115 @@ impl CompactionJob {
     }
 }
 
+/// Commit update for a given partition.
+#[derive(Debug)]
+pub struct CommitUpdate {
+    /// Partition to be updated.
+    pub(crate) partition_id: PartitionId,
+    /// Files to be deleted.
+    pub(crate) delete: Vec<ParquetFile>,
+    /// Files to be upgraded.
+    pub(crate) upgrade: Vec<ParquetFile>,
+    /// Target level for upgraded files.
+    pub(crate) target_level: CompactionLevel,
+    /// Files to be created.
+    pub(crate) create: Vec<ParquetFileParams>,
+}
+
+impl CommitUpdate {
+    /// Create new commit update.
+    pub fn new(
+        partition_id: PartitionId,
+        delete: Vec<ParquetFile>,
+        upgrade: Vec<ParquetFile>,
+        create: Vec<ParquetFileParams>,
+        target_level: CompactionLevel,
+    ) -> Self {
+        Self {
+            partition_id,
+            delete,
+            upgrade,
+            target_level,
+            create,
+        }
+    }
+}
+
+/// Reason for skipping a partition.
+#[derive(Debug)]
+pub enum SkipReason {
+    /// Partition is not compactible, due to an encountered error.
+    CompactionError(String),
+}
+
+/// Status.
+#[derive(Debug)]
+pub enum CompactionJobStatusVariant {
+    /// Updates associated with ongoing compaction job.
+    Update(CommitUpdate),
+    /// Request to skip partition.
+    RequestToSkip(SkipReason),
+    /// Compaction job is complete.
+    Complete,
+    /// Compaction job has failed.
+    Error(ErrorKind),
+}
+
+/// Status ([`CompactionJobStatusVariant`]) associated with a [`CompactionJob`].
+#[derive(Debug)]
+pub struct CompactionJobStatus {
+    /// Job.
+    pub job: CompactionJob,
+    /// Status.
+    pub status: CompactionJobStatusVariant,
+}
+
+/// Status of a compaction job.
+#[derive(Debug)]
+pub enum CompactionJobStatusResult {
+    /// Ack only.
+    Ack,
+    /// Updates which were processed.
+    UpdatedParquetFiles(Vec<ParquetFileId>),
+}
```

Comment:
When I see a type that ends in the name `Result`, I assume it's going to have variants similar to `std::result::Result`'s `Ok` or `Err`, probably a type alias for `std::result::Result` with a default of a specialized `E` type. 

I know that's not the general meaning of "result" in English, but it's a fairly consistent convention in Rust :) 

What do you think about changing the name to something like `CompactionJobStatusResponse`? And actually, this is what the scheduler is going to send back to the compactors to say that the scheduler has processed the job status update that the compactor sent the scheduler, right? So I think the docs need to be updated here too, because it's not the actual status of the compaction job, it's the status of the status update 🧟‍♀️ 

So how about something like...

```suggestion
/// Responses sent from the scheduler to the compactor when the compactor sends an updated job status.
#[derive(Debug)]
pub enum JobStatusUpdateResponse {
    /// Ack only.
    Ack,
    /// Updates which were processed.
    UpdatedParquetFiles(Vec<ParquetFileId>),
}
```

Reply:
Yeah, better words are definitely needed! 🤦🏼‍♀️